beaker_kernel/lib/llm_tasks.py

```diff
@@ -0,0 +1,57 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any
+from enum import Enum
+import asyncio
+import json
+
+from archytas.agent import Agent
+
+OUTPUT_CHAR_LIMIT = 1000
+
+async def summarize(notebook: dict,
+    summary_types=[
+        "a single sentence BLUF in past tense imperative",
+        "a three sentence summary",
+        "a paragraph summary",
+    ]
+):
+    for cell in notebook["cells"]:
+        if "outputs" in cell and len(cell["outputs"]) > 0:
+            for output in cell["outputs"]:
+                if "text" in output:
+                    output["text"] = output["text"][:OUTPUT_CHAR_LIMIT]
+                if "traceback" in output:
+                    del output["traceback"]
+                if "data" in output:
+                    for data_type in output["data"]:
+                        output["data"][data_type] = output["data"][data_type][:OUTPUT_CHAR_LIMIT]
+            
+    agent = Agent()
+    history_prompt = f"""
+You will generate a JSON array containing summaries of the events from the LLM-powered notebook below.
+
+Only respond with a JSON array. Do not include backticks '`' or extra text. 
+
+{notebook}
+"""
+
+    history_query = """
+Generate a list where each element is a summary of each event in an imperative mood, 
+e.g. "Load dataset of Indonesian population from 2003 to 2006".
+    """
```

Comment:
Any reason for this grammatical mood selection?
In my head it was always in the past-tense indicative mood.
"The Indonesian population dataset from 2003 to 2006 was loaded."
or something like that.

Also, I think the we might get better results if we try to decrease some vagueness in the language here.
> Generate a list where each element is

vs
> Generate a list consisting of elements which each contain

and
> each element is a summary of each event

vs
> each element summarizes a cell in the notebook provided above

Reply:
I think we want to use active voice for summaries because it usually is clearer and terser. Imperative mood is active and allows us to remove details like whether the _agent_ answered the question or the _user_ ran the code. I was considering making the elements past tense so "Load_ed_ Indonesian population dataset from 2003 to 2006". Good commit messages are oftentimes imperative as well because it's immediately clear what action was taken and cuts out unnecessary words. Unless you're programming with Forth or using a binary operator, the function call (action) precedes the elements it operates on. When describing events, this seemed like the obvious choice to me at the time

Reply:
If we use indicative mood without saying "the user" or "the agent", we'll end up getting passive voice

Reply:
closing this issue for now because now the agent has access to the notebook and the list